lib/src/document.dart

```diff
@@ -84,6 +87,41 @@ class Document {
       }
     }
   }
+
+  static Element _classElement(
+          String tag, String klass, List<Node>? children) =>
+      Element(tag, children)..attributes['class'] = klass;
+
+  void _filterFootnotes(List<Node> nodes) {
+    final footnotes = <Element>[];
+    final blocks = <Node>[];
+
+    for (final node in nodes) {
+      if (node is Element &&
+          node.tag == 'li' &&
+          node.attributes['class'] == 'footnote-item') {
+        final label = node.attributes.remove('label');
+        if (label != null && (footnoteReferences[label] ?? 0) > 0) {
+          footnotes.add(node);
+        }
+      } else {
+        blocks.add(node);
+      }
+    }
+    nodes
+      ..clear()
+      ..addAll(blocks);
+
+    final section = footnotes.isEmpty
```

Comment:
Can these few lines be simplified by guarding them all with `footnotes.isEmpty` rather than doing a second guard below with `section != null`.

Reply:
change guarding to `isEmpty`